src/components/templates/PartyMap/components/Map/Map.tsx

```diff
@@ -1,197 +1,175 @@
 
-interface PropsType {
+interface MapProps {
+  user: FirebaseReducer.AuthState;
+  profileData: UserExperienceData;
   venue: PartyMapVenue;
-  attendances: { [location: string]: number };
+  partygoers: readonly WithId<User>[];
   selectedRoom: PartyMapRoomData | undefined;
-  setSelectedRoom: (room: PartyMapRoomData | undefined) => void;
-  setIsRoomModalOpen: (value: boolean) => void;
+  selectRoom: (room: PartyMapRoomData) => void;
+  unselectRoom: () => void;
+  enterSelectedRoom: () => void;
 }
 
-const DEFAULT_COLUMNS = 40;
-const DEFAULT_ROWS = 25;
-
-export const Map: React.FC<PropsType> = ({
+export const Map: React.FC<MapProps> = ({
+  user,
+  profileData,
   venue,
-  attendances,
+  partygoers,
   selectedRoom,
-  setSelectedRoom,
-  setIsRoomModalOpen,
+  selectRoom,
+  unselectRoom,
+  enterSelectedRoom,
 }) => {
   const venueId = venue.id;
-  const { user, profile } = useUser();
-  const [selectedUserProfile, setSelectedUserProfile] = useState<
-    WithId<User>
-  >();
-  const [rows, setRows] = useState<number>(0);
-
-  const columns = venue.columns ?? DEFAULT_COLUMNS;
-  const currentPosition = profile?.data?.[venue.id];
+  const userUid = user?.uid;
+  const showGrid = venue.showGrid;
 
-  const columnsArray = useMemo(() => Array.from(Array<JSX.Element>(columns)), [
-    columns,
-  ]);
-  const rowsArray = useMemo(() => Array.from(Array(rows)), [rows]);
+  const totalColumns = venue.columns ?? DEFAULT_COLUMNS;
+  const [totalRows, setTotalRows] = useState<number>(0);
 
-  const venues = useSelector(orderedVenuesSelector);
-  const partygoers: readonly WithId<User>[] | undefined = useSelector(
-    partygoersSelector
+  const columnsArray = useMemo(
+    () => Array.from(Array<JSX.Element>(totalColumns)),
+    [totalColumns]
   );
+  const rowsArray = useMemo(() => Array.from(Array(totalRows)), [totalRows]);
 
   useEffect(() => {
     const img = new Image();
     img.src = venue.mapBackgroundImageUrl ?? "";
     img.onload = () => {
       const imgRatio = img.width ? img.width / img.height : 1;
+
       const calcRows = venue.columns
         ? Math.round(parseInt(venue.columns.toString()) / imgRatio)
         : DEFAULT_ROWS;
-      setRows(calcRows);
+
+      setTotalRows(calcRows);
     };
   }, [venue.columns, venue.mapBackgroundImageUrl]);
 
+  const takeSeat = useCallback(
+    (row: number | null, column: number | null) => {
+      makeUpdateUserGridLocation({
+        venueId,
+        userUid: user?.uid,
+        profileData,
+      })(row, column);
+    },
+    [profileData, user, venueId]
+  );
+
+  const currentPosition = profileData?.[venue.id];
+
+  const checkForRoomHit = useCallback(
+    (row: number, column: number) => {
+      const roomHitFilter = makeRoomHitFilter({
+        row,
+        column,
+        totalRows,
+        totalColumns: totalColumns,
+      });
+
+      // Only select the first room if we hit multiple (eg. overlapping)
+      const roomHit = venue.rooms.find(roomHitFilter);
```

Comment:
Rooms can only hit here if they are overlapping the same bubble that the user 'is standing on'. So at that point, they would both be equidistant anyway.

IMO the better solution here is pushing validation higher to prevent rooms from being able to overlap at all.